src/classes/RD2_RecurringDonations_TEST.cls

```diff
@@ -332,34 +334,66 @@ private with sharing class RD2_RecurringDonations_TEST {
     }
 
     /***
-    * @description Verifies that start date is updated to current date on updated RD when appropriate
+    * @description Verifies that start date is NOT updated to current date on updated RD when appropriate
     */
     @isTest
-    private static void shouldOverrideStartDateWithCurrentDate() {
-        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
-
-        Contact contact = getContact();
-
-        npe03__Recurring_Donation__c rd = TEST_RecurringDonationBuilder.constructEnhancedBuilder()
-            .withContact(contact.Id)
-            .withAmount(50)
-            .withInstallmentPeriod(RD2_Constants.INSTALLMENT_PERIOD_MONTHLY)
-            .withDayOfMonth('10')
-            .withInstallmentFrequency(1)
-            .withStatus(RD2_Constants.STATUS_ACTIVE)
-            .withDateEstablished(Date.newInstance(2019, 6, 5))
-            .withStartDate(Date.newInstance(2019, 6, 5))
+    private static void shouldNotOverrideStartDateWithCurrentDate() {
+        Date startDate = Date.newInstance(2019, 10, 5);
+        npe03__Recurring_Donation__c rd = getRecurringDonationMonthlyBuilder()
```

Comment:
I'd like to understand the best practice we are targeting.  We have refactored other tests significantly, so I'd like to understand what drives the distinction between the two approaches.

Reply:
New helper methods are applied to changed tests only although they might be used in more tests. Most importantly, since this is a critical change, less moving parts makes it easier to determine existing tests work as before for the same RD condition. I am additionally making changes to existing tests in PR #4969  and there might be duplication of the effort.